components/infobox/commons/infobox_weapon.lua

```diff
@@ -0,0 +1,108 @@
+-- @Liquipedia
+-- wiki=commons
+-- page=Module:Infobox/Weapon
+--
+-- Please see https://github.com/Liquipedia/Lua-Modules to contribute
+--
+
+local Class = require('Module:Class')
+local Namespace = require('Module:Namespace')
+local BasicInfobox = require('Module:Infobox/Basic')
+local Flags = require('Module:Flags')
+
+local Widgets = require('Module:Infobox/Widget/All')
+local Cell = Widgets.Cell
+local Header = Widgets.Header
+local Title = Widgets.Title
+local Center = Widgets.Center
+local Customizable = Widgets.Customizable
+
+local Weapon = Class.new(BasicInfobox)
+
+function Weapon.run(frame)
+	local weapon = Weapon(frame)
+	return weapon:createInfobox()
+end
+
+function Weapon:createInfobox()
+	local infobox = self.infobox
+	local args = self.args
+
+	local widgets = {
+	Header{
+		name = self:nameDisplay(args),
+		image = args.image,
+		imageDefault = args.default,
+		imageDark = args.imagedark or args.imagedarkmode,
+		imageDefaultDark = args.defaultdark or args.defaultdarkmode,
+	},
+	Center{content = {args.caption}},
+	Title{name = (args.informationType or 'Weapon') .. ' Information'},
+	Cell{
+		name = 'Class',
+		content = self:getAllArgsForBase(args, 'class', {makeLink = true}),
+	},
+	Cell{
+		name = 'Origin',
+		content = {self:_createLocation(args.origin)},
+	},
+	Cell{name = 'Price', content = {args.price}},
+	Cell{name = 'Kill Award', content = {args.killaward}},
+	Cell{name = 'Base Damage', content = {args.damage}},
+	Cell{name = 'Magazine Size', content = {args.magsize}},
+	Cell{name = 'Ammo Capacity', content = {args.ammocap}},
+	Cell{name = 'Reload Speed', content = {args.reloadspeed}},
+	Cell{name = 'Rate of Fire', content = {args.rateoffire}},
+	Cell{name = 'Firing Mode', content = {args.firemode}},
+	Customizable{
+		id = 'side',
+		children = {
+			Cell{name = 'Side', content = {args.side}},
+		}
+	},
+	Customizable{
+		id = 'user',
+		children = {
+			Cell{
+				name = 'User(s)',
```

Comment:
Can we make the *s* dependant on number of users?

Reply:
In your opinion, would it be better to just drop the parenthesis and have it be users for all cases, or make the s dependent on the number of users? Looking at the other base infoboxes on commons, it seems the first option i listed is preferred

Reply:
I think having it dependent is better.
See Organizers in Infobox League for an example